client/asset/firo/block_deserialize.go

```diff
@@ -0,0 +1,205 @@
+package firo
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"math"
+	"time"
+
+	"decred.org/dcrdex/client/asset/btc"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// const (
+// 	// Only blocks mined with progpow are considered.
+// 	// Previous mining algorithms: MTP and Lyra2z ignored as being too early for
+// 	// Firo wallet on Dex ~ late 2023
+// 	ProgpowStartTime        = 1635228000 // Tue Oct 26 2021 06:00:00 UTC+0
+// 	HeaderLength            = 80
+// 	ProgpowExtraLength      = 40
+// 	ProgpowFullHeaderLength = HeaderLength + ProgpowExtraLength
+// )
+
+type firoBlock struct {
+	hdrHash chainhash.Hash
+	hdr     *wire.BlockHeader
+	txns    []*wire.MsgTx
+}
+
+// deserializeBlock hashes the header bytes from the serialized Firo block passed
+// in blk. Based on block hash re-gets the block as json and returns the header
+// plus any Transparent transactions found parsed into a wire.MsgBlock.
+//
+// The other 10 transaction types are discarded; including coinbase.
+func getBlock(c btc.RpcCaller, hash chainhash.Hash) (*wire.MsgBlock, error) {
+	// func deserializeBlock(c rpcCaller, net *chaincfg.Params, blk []byte) (*wire.MsgBlock, error) {
+	firoBlock := &firoBlock{}
+
+	// // hash header
+	// var header []byte
+	// switch net.Name {
+	// case "mainnet", "testnet3":
+	// 	header = make([]byte, ProgpowFullHeaderLength)
+	// 	copy(header, blk[:ProgpowFullHeaderLength])
+	// case "regtest":
+	// 	header = make([]byte, HeaderLength)
+	// 	copy(header, blk[:HeaderLength])
+	// default:
+	// 	return nil, fmt.Errorf("unknown net: %s", net.Name)
+	// }
+	// firoBlock.hdrHash = chainhash.DoubleHashH(header)
+
+	// fmt.Printf("firo deserialized block hash: %s\n", firoBlock.hdrHash.String())
+
+	// get json block
+	jsonBlock, err := getJsonBlock(c, firoBlock.hdrHash)
```

Comment:
```suggestion
	jsonBlock, err := getJsonBlock(c, hash)
```

`firoBlock` is not needed.

Reply:
I will play with this .. but am in and out of the country at the moment